File: mobile/src/main/java/com/windscribe/mobile/splittunneling/SplitTunnelingPresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.splittunneling;

import android.content.Context;

public interface SplitTunnelingPresenter {

    void onBackPressed();

    void onDestroy();

    void onFilter(String query);

    void onNewRoutingModeSelected(String mode);

    void onToggleButtonClicked();

    void setTheme(Context context);

    void setupLayoutBasedOnPreviousSettings();
}

File: mobile/src/main/java/com/windscribe/mobile/splittunneling/SplitTunnelingPresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.splittunneling

import android.content.Context

interface SplitTunnelingPresenter {
    fun onBackPressed()
    fun onDestroy()
    fun onFilter(query: String)
    fun onNewRoutingModeSelected(mode: String)
    fun onToggleButtonClicked()
    fun setTheme(context: Context)
    fun setupLayoutBasedOnPreviousSettings()
}